### PR TITLE
Authentication with next-auth

### PR DESCRIPTION
**Pull request by muhammedshamil8**
- Implemented user authentication functionality.
- Set up registration and login forms.
- Configured middleware for protecting logged-in dashboard pages.
- Designed UI components using atomic design principles.
- Initial UI created using atom design structure components.
- Planning to transition to Material UI for improved styling and functionality.

### CODE REVIEW

**Comment by mohammedafeef** [comment]
@muhammedshamil8 Please resolve the conflicts

File: README.md
Comment (mohammedafeef) [inline README.md:1]: praise: doc update will help other better understand the project

File: app/Providers.js
Comment (mohammedafeef) [inline app/Providers.js:1]: Question: Could you tell me how this auth session works? is it something like a context provider? please take it as a humble doubt

**Review by mohammedafeef — APPROVED** [review]

File: README.md
Comment (muhammedshamil8) [reply README.md:1]: okk sure i will update tonight

File: app/Providers.js
Comment (muhammedshamil8) [reply app/Providers.js:1]: ### Explanation of Authentication Session in our Project

In our project, we utilize NextAuth.js to manage the authentication session. While it doesn't function exactly like a context provider, it serves a similar purpose by handling user authentication across the application.

#### Key Features:
- **Providers:** We use `CredentialsProvider` from NextAuth.js, which allows users to authenticate using their username and password.
- **Database Integration:** NextAuth.js seamlessly connects with our MongoDB database to store and retrieve user information securely.
- **Session Management:** Authentication sessions are managed using JWT (JSON Web Tokens), ensuring that users remain authenticated as they navigate through our application.

NextAuth.js simplifies the login and registration processes and ensures robust security measures are in place for managing user sessions effectively.

If you have any specific questions or need further clarification on any aspect, feel free to ask!

**Pull request merged**